Signal/src/AppDelegate.m

```diff
@@ -172,10 +172,13 @@ - (UIViewController *)loadingRootViewController
     BOOL shouldShowUpgradeLabel = NO;
     NSString *lastAppVersion = AppVersion.instance.lastAppVersion;
     NSString *lastCompletedLaunchAppVersion = AppVersion.instance.lastCompletedLaunchAppVersion;
+    // Every time we change a database view in such a way that might cause a delay on launch,
+    // we need to bump this constant.
+    NSString *lastVersionWithDatabaseViewChange = @"2.13.0";
     BOOL mayNeedUpgrade = ([TSAccountManager isRegistered] &&
-        [VersionMigrations isVersion:lastAppVersion atLeast:@"2.0.0" andLessThan:@"2.13.0"]);
+        [VersionMigrations isVersion:lastAppVersion atLeast:@"2.0.0" andLessThan:lastVersionWithDatabaseViewChange]);
     BOOL hasCompletedUpgrade = (lastCompletedLaunchAppVersion &&
-        [VersionMigrations isVersion:lastCompletedLaunchAppVersion atLeast:@"2.13.0"]);
+        [VersionMigrations isVersion:lastCompletedLaunchAppVersion atLeast:lastVersionWithDatabaseViewChange]);
 
     // We added a number of database views in v2.13.0.
     if (mayNeedUpgrade && !hasCompletedUpgrade) {
```

Comment:
I may be missing some of the nuance here, but can we simplify this to something like:

```
BOOL shouldShowUpgradeLabel = [TSAccountManager isRegistered] && lastCompletedLaunchAppVersion && ![VersionMigrations isVersion:lastCompletedLaunchAppVersion atLeast:lastVersionWithDatabaseViewChange]
```

I believe the only affect of:
```
[VersionMigrations isVersion:lastAppVersion atLeast:@"2.0.0" andLessThan:lastVersionWithDatabaseViewChange]);
```

would be to omit the label if the app didn't complete upgrading the last time it launched this version. Which seems undesirable.